include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -7082,7 +7082,7 @@ namespace VectorTools
         double my_values[3] = { mean_double.real(), mean_double.imag(), area };
         double global_values[3];
 
-        MPI_Allreduce (&my_values, &global_values, 3, MPI_DOUBLE,
+        MPI_Allreduce (my_values, global_values, 3, MPI_DOUBLE,
```

Comment:
Indeed, that was a scary bug. Thanks for fixing it. (Could this have been compiler-specific behavior that worked on some machines or was this never tested?)